kube-runtime/src/controller/mod.rs

```diff
@@ -277,31 +298,58 @@ where
 /// ```
 pub struct Controller<K>
 where
-    K: Clone + Meta + Debug + 'static,
+    K: Clone + Meta + 'static,
+    <K as Meta>::Family: Debug + Eq + Hash + Debug + Clone,
 {
     // NB: Need to Unpin for stream::select_all
     // TODO: get an arbitrary std::error::Error in here?
     selector: SelectAll<BoxStream<'static, Result<ObjectRef<K>, watcher::Error>>>,
+    family: K::Family,
     reader: Store<K>,
 }
 
 impl<K> Controller<K>
 where
-    K: Clone + Meta + DeserializeOwned + Debug + Send + Sync + 'static,
+    K: Clone + Meta + DeserializeOwned + Send + Sync + 'static,
+    <K as Meta>::Family: Debug + Eq + Hash Debug + Clone + Default,
 {
     /// Create a Controller on a type `K`
     ///
     /// Configure `ListParams` and `Api` so you only get reconcile events
     /// for the correct `Api` scope (cluster/all/namespaced), or `ListParams` subset
     #[must_use]
     pub fn new(owned_api: Api<K>, lp: ListParams) -> Self {
-        let writer = Writer::<K>::default();
+        Self::new_with(owned_api, lp, Default::default())
+    }
+}
+
+impl<K> Controller<K>
+where
+    K: Clone + Meta + DeserializeOwned + Send + Sync + 'static,
+    <K as Meta>::Family: Debug + Eq + Hash + Clone,
```

Comment:
Are you able to help me summarise where the various requiments on the Family parameter comes from?

Reply:
As far as I remember, these are traits implemented by ObjectRef.